src/System.IO/src/System/IO/Stream.cs

```diff
@@ -408,25 +408,24 @@ public override void Flush()
             {
             }
 
-#pragma warning disable 1998 // async method with no await
-            public override async Task FlushAsync(CancellationToken cancellationToken)
+            public override Task FlushAsync(CancellationToken cancellationToken)
             {
-                cancellationToken.ThrowIfCancellationRequested();
+                return cancellationToken.IsCancellationRequested ?
+                    Task.FromCanceled(cancellationToken) :
+                    Task.CompletedTask;
             }
-#pragma warning restore 1998
 
             public override int Read(byte[] buffer, int offset, int count)
             {
                 return 0;
             }
 
-#pragma warning disable 1998 // async method with no await
-            public override async Task<int> ReadAsync(Byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+            public override Task<int> ReadAsync(Byte[] buffer, int offset, int count, CancellationToken cancellationToken)
             {
-                cancellationToken.ThrowIfCancellationRequested();
-                return 0;
+                return cancellationToken.IsCancellationRequested ?
+                    Task.FromCanceled<int>(cancellationToken) :
+                    Task.FromResult(0); // TODO: Get this Task from a cache for better perf
```

Comment:
This change is making it worse: 0 being returned was already resulting in a cached task being returned, and now it's not.